lib/repo/mapping.go

```diff
@@ -0,0 +1,150 @@
+package repo
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+const missingRepo = "Device repo is missing"
+
+var path string
+
+func init() {
+	//p, err := os.Executable()
+	//if err != nil {
+	//	logrus.Error(err)
+	//}
+
+	path = filepath.Join(baseDir, file)
+}
+
+// DeviceImage contains url, title, username and password which are used after flashing
+type DeviceImage struct {
+	Url   string `json`
+	Title string `json`
+	User  string `json`
+	Pass  string `json`
+}
+
+// DeviceMapping is a collection of device, it sub-types and sets of images for these devices
+type DeviceMapping struct {
+	Name   string           `json`
+	Sub    []*DeviceMapping `json`
+	Images []DeviceImage    `json`
+
+	dir string
+	Url DeviceImage
+}
+
+// GetSubsNames returns array of Names within a `Sub`
+func (d *DeviceMapping) GetSubsNames() []string {
+	r := make([]string, len(d.Sub))
+	for i, o := range d.Sub {
+		r[i] = o.Name
+	}
+
+	return r
+}
+
+// GetImageTitles returns array of image titles
+func (d *DeviceMapping) GetImageTitles() []string {
+	r := make([]string, len(d.Images))
+	for i, o := range d.Images {
+		r[i] = o.Title
+	}
+
+	return r
+}
+
+// Dir returns directory of a local repo `.iotit/images/{name}`
+func (d *DeviceMapping) Dir() string {
+	return filepath.Join(ImageDir, d.dir)
+}
+
+// deviceCollection is a starting point of the collection of images
+type deviceCollection struct {
+	Devices []*DeviceMapping `json`
```

Comment:
__[gometalinter]__
* [vet] struct field tag `json` not compatible with reflect.StructTag.Get: bad syntax for struct tag pair

<!-- Comment by [SideCI](https://sideci.com) -->